File: VB/GridView.Scrolling/Controllers/GridViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports GridView.Scrolling.Models

Namespace GridView.Scrolling.Controllers
	<HandleError> _
	Public Class GridViewController
		Inherits Controller
		Public Function Index() As ActionResult
			Return Scrolling()
		End Function
		Public Function Scrolling() As ActionResult
			ViewData("ShowVerticalScrollBar") = True
			ViewData("ShowHorizontalScrollBar") = True
			Return View("Scrolling", NorthwindDataProvider.GetInvoices())
		End Function
		Public Function ScrollingPartial() As ActionResult
			GetRequestParemetrs()
			Return PartialView("ScrollingPartial", NorthwindDataProvider.GetInvoices())
		End Function
		Private Sub GetRequestParemetrs()
			ViewData("ShowVerticalScrollBar") = If(String.IsNullOrEmpty(Request.Params("ShowVerticalScrollBar")), True, Boolean.Parse(Request.Params("ShowVerticalScrollBar")))
			ViewData("ShowHorizontalScrollBar") = If(String.IsNullOrEmpty(Request.Params("ShowHorizontalScrollBar")), True, Boolean.Parse(Request.Params("ShowHorizontalScrollBar")))
		End Sub
	End Class
End Namespace

File: CS/GridView.Scrolling/Controllers/GridViewController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using GridView.Scrolling.Models;

namespace GridView.Scrolling.Controllers {
    [HandleError]
    public class GridViewController : Controller {
        public ActionResult Index() {
            return Scrolling();
        }
        public ActionResult Scrolling(){
            ViewData["ShowVerticalScrollBar"] = true;
            ViewData["ShowHorizontalScrollBar"] = true;
            return View("Scrolling", NorthwindDataProvider.GetInvoices());
        }
        public ActionResult ScrollingPartial() {
            GetRequestParemetrs();
            return PartialView("ScrollingPartial", NorthwindDataProvider.GetInvoices());
        }
        void GetRequestParemetrs() {
            ViewData["ShowVerticalScrollBar"] = string.IsNullOrEmpty(Request.Params["ShowVerticalScrollBar"]) ? true : bool.Parse(Request.Params["ShowVerticalScrollBar"]);
            ViewData["ShowHorizontalScrollBar"] = string.IsNullOrEmpty(Request.Params["ShowHorizontalScrollBar"]) ? true : bool.Parse(Request.Params["ShowHorizontalScrollBar"]);
        }
    }
}
